src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +42,1136 @@ const ActorInit En_Niw_InitVars = {
+        }
+        if (this->actionFunc != func_80AB6BF8 && this->actionFunc != func_80AB6D08 &&
+            this->actionFunc != func_80AB6324 && this->actionFunc != func_80AB63A8 &&
+            this->actionFunc != func_80AB6450) {
+            CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6BF8.s")
+s32 EnNiw_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnNiw* this = THIS;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6D08.s")
+    if (limbIndex == 13) {
+        rot->y += (s16)this->unk_2DC;
+    }
+    if (limbIndex == 15) {
+        rot->y += (s16)this->unk_2E0;
+    }
+    if (limbIndex == 11) {
+        rot->x += (s16)this->unk_2D8;
+        rot->y += (s16)this->unk_2D4;
+        rot->z += (s16)this->unk_2D0;
+    }
+    if (limbIndex == 7) {
+        rot->x += (s16)this->unk_2CC;
+        rot->y += (s16)this->unk_2C8;
+        rot->z += (s16)this->unk_2C4;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6EB4.s")
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB6F04.s")
+void EnNiw_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiw* this = THIS;
+    Vec3f scale = { 0.15f, 0.15f, 0.15f };
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB70A0.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                     EnNiw_OverrideLimbDraw, NULL, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB70F8.s")
+    if (this->actionFunc == func_80AB6450) {
+        func_80033C30(&this->actor.posRot.pos, &scale, 255, globalCtx);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB714C.s")
+    EnNiw_ParticleDraw(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB7204.s")
+void EnNiw_ParticleSpawn(EnNiw* this, Vec3f* pos, Vec3f* vel, Vec3f* accel, f32 scale) {
+    s16 i;
+    EnNiwParticle* particle = this->particle;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw/func_80AB7290.s")
+    for (i = 0; i < ARRAY_COUNT(this->particle); i++, particle++) {
+        if (particle->type == 0) {
+            particle->type = 1;
+            particle->pos = *pos;
+            particle->vel = *vel;
+            particle->accel = *accel;
+            particle->timer = 0;
+            particle->scale = scale / 1000.0f;
+            particle->lifespan = (s16)Math_Rand_ZeroFloat(20.0f) + 40;
+            particle->unk_2A = Math_Rand_ZeroFloat(1000.0f);
+            return;
```

Comment:
It's not a huge deal, but I'd prefer `break` to `return` here since the purpose of this statement is to exit the loop rather than the entire function. It just so happens that the function ends here.